src/FixedDataTable.js

```diff
@@ -576,10 +576,6 @@ class FixedDataTable extends React.Component {
       this.props.scrollLeft !== nextProps.scrollLeft) {
       this._didScrollStart();
     }
-
-    // Cancel any pending debounced scroll handling and handle immediately.
-    this._didScrollStop.reset();
-    this._didScrollStopSync();
```

Comment:
I'm not sure, but I'm planning to test that more before merging.  My hope is since scrolling is no longer entangled with the setState method, calling _didScrollStop should no longer interfere with the scroll to row.  There could be a risk though if:

1. The redux store is updated by the scrollToRow pro
2. The async call to didScrollStop triggers onScrollEnd with the previous scrollX
3. The component is controlled with the old scrollX and overrides the scrollToRow prop

I'll do what I can to test for that case.  I'll also see if I can remove the _isScrolling variable and replace it with the scrolling value from the store.  That may offer a chance to move onScrollEnd's logic into an action and keep it in sync with the store...

Reply:
> I'll also see if I can remove the _isScrolling variable and replace it with the scrolling value from the store.

I think we can just use `props.isScrolling` instead of `this._isScrolling`.
They are both set/unset at the same place. See [here](https://github.com/schrodinger/fixed-data-table-2/blob/v1.0-beta/src/FixedDataTable.js#L1053) and [here](https://github.com/schrodinger/fixed-data-table-2/blob/v1.0-beta/src/FixedDataTable.js#L1077).
And the `isScrolling` prop seems to be unused after the reducer [changes the state](https://github.com/schrodinger/fixed-data-table-2/blob/v1.0-beta/src/reducers/index.js#L143).

Reply:
> 1. The redux store is updated by the scrollToRow pro
> 2. **The async call to didScrollStop triggers onScrollEnd with the previous scrollX**
> 3. The component is controlled with the old scrollX and overrides the scrollToRow prop

(Correct me if I'm wrong) Does `onScrollEnd` get the previous `scrollX`, _if_ FixedDataTable wasn't rendered after the update (step 1). 
Then it uses the stale prop, which will then be used to _just_ dispatch the old `scrollX` value to `props.onScrollEnd` and stop scrolling?

I think where I'm confused is I couldn't see where our component is _re-updated_ with the old `scrollX` due to [_didScrollStopSync](https://github.com/schrodinger/fixed-data-table-2/blob/v1.0-beta/src/FixedDataTable.js#L1064).

Reply:
I tried testing this and couldn't reproduce any issues.  I'm going to close this review in favor of yours since you've got a bit more worked out than me.